compiler/qsc/src/incremental.rs

```diff
@@ -140,6 +179,52 @@ impl Compiler {
         Ok(increment)
     }
 
+    /// Compiles Q# ast fragments. See [`compile_ast_fragments_fail_fast`] for more details.
+    ///
+    /// This method calls an accumulator function with any errors returned
+    /// from each of the stages (parsing, lowering).
+    /// If the accumulator succeeds, compilation continues.
+    /// If the accumulator returns an error, compilation stops and the
+    /// error is returned to the caller.
+    pub fn compile_ast_fragments<F>(
+        &mut self,
+        source_name: &str,
+        source_contents: &str,
```

Comment:
I think it's worth a clarifying comments here just to make sure it's understood that these are use for error reporting and not themselves compiled.

Reply:
Will update when the PR using the API is added.